### PR TITLE
Sorting of versions

### PR DESCRIPTION
**Pull request by galaplexus**
Apologies for the if { for { if { if { if {
This is actually my first time coding javascript and I didn't manage to split in into 2 functions into this export object.
If someone want to re-work this, I'd be very glad to see how you do it.

### CODE REVIEW

**Comment by randombenj** [comment]
@fliiiix and I will have a look at this after fixing: https://github.com/docat-org/docat/issues/137 and updating the js deps: https://github.com/docat-org/docat/pull/141

File: web/src/main.js
Comment (lukasweber) [inline web/src/main.js:62]: You may want to revert this change now that the fix for #137 has been merged. Furthermore, the `latest` tag is nothing which is currently really "known" or enforced by the logic. You still have to create the `latest` tag manually with e.g. : `curl -X PUT http://localhost:8000/api/PROJECT/VERSION/tags/latest`

**Comment by fliiiix** [comment]
With all the bug out of the way what is the way forward here? 
@randombenj do you want to review it first or should we rebase it first to master?

**Comment by galaplexus** [comment]
My fix suggestion is reverted. All left is the sorting

**Comment by fliiiix** [comment]
LGTM i guess a review from @randombenj would be nice since im not a javascript dev and then we can merge this

**Review by fliiiix — APPROVED** [review]
commits are fine for me - but @randombenj should review the actual js implementation

**Comment by lukasweber** [comment]
@fliiiix Looks good to me :) I just wonder, do you want to have as few dependencies as possible to third-party packages? Otherwise it could be worth including [semver](https://www.npmjs.com/package/semver) and maybe add an additional rule for `latest` on top. So, you won't have to include custom logic with tests for it.

**Comment by randombenj** [comment]
@galaplexus Thanks for the contribution, especially the tests are highly appreciated! :tada:  Before merging a few general things:

- [ ] please use `camelCase` in the js code so we keep it consistant. (we might need a linter in the future)
- [ ] as @lukasweber mentioned, please use [semver](https://github.com/npm/node-semver) to parse the versions.

So in general I would propose to do this:

1) (depending on discussion mentioned below) If one of both is `latest` this is greater
1) Use `semver` to try and parse both versions (compare them if both are parsable)
1) If one of both is not parsable `semver.valid('a.b.c') // null` then do plain string comparison

@fliiiix @timofurrer @lukasweber Until now `latest` is just a general tag, this change proposes to 'hardcode' latest as the default version. I personally am fine with this, however we should document the behaviour to make it clear that `latest` is a special tag.

**Comment by fliiiix** [comment]
Im not a huge fan of hardcoding the `latest` tag but I guess its fine for now

**Comment by randombenj** [comment]
@fliiiix Me neither, but it's already the default redirect so I guess we can just keep the behaviour and at least document it.

**Comment by galaplexus** [comment]
Thx for feedback.
Changed accordingly.
One test had to change. This is debatable but whatever - side case.

On the topic of latest keyword. My suggestion is to reserve this tag and assign it automatically according to the semver logic. This would be done in the python API on post and delete. Then #40 can be handled easily as well

**Comment by randombenj** [comment]
@galaplexus looks much better thanks for the improvement, I would suggest getting rid of the log statements before merging?
I think the `latest` handling must be discussed some more in the future ...

**Comment by fliiiix** [comment]
also commits need to be cleaned up before merging

**Comment by galaplexus** [comment]
Don't know exactly what happened but now the web/src/main.js is exactly like on your main (somehow still shows as a diff.

**Comment by randombenj** [comment]
@galaplexus I think you might have pushed some old version of the branch again?

**Comment by fliiiix** [comment]
I guess you should rebase (https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) your branch on the latest `main` and instead of reverting commits you should remove them from this PR.

**Comment by galaplexus** [comment]
Apologies. Started using GitHub desktop and my branch was not in sync

**Comment by randombenj** [comment]
Great thanks for the fix!

**Comment by fliiiix** [comment]
reverted see https://github.com/docat-org/docat/pull/167

**Pull request merged**
